### PR TITLE
update release 3.1

### PR DESCRIPTION
**Pull request by overvenus**
Merge recent commits into release-3.1

### CODE REVIEW

File: tests/br_incremental/run.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# incremental backup
+echo "incremental backup start..."
+run_br --pd $PD_ADDR backup table -s "local://$TEST_DIR/$DB" --db $DB -t $TABLE --ratelimit 5 --concurrency 4 --lastbackupts $last_backup_ts
+
+start_ts=$(br validate decode --field="start-version" -s "local://$TEST_DIR/$DB" | tail -n1)
+end_ts=$(br validate decode --field="end-version" -s "local://$TEST_DIR/$DB" | tail -n1)
+
+echo "start version: $start_ts, end version: $end_ts"
+
+# incremental restore
```
Comment (3pointer) [inline tests/br_incremental/run.sh]: It seems we did not cleanup data after Line43, so it always success even though restore failed?

File: pkg/restore/import.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/pingcap/kvproto/pkg/kvrpcpb"
 	"github.com/pingcap/log"
 	restore_util "github.com/pingcap/tidb-tools/pkg/restore-util"
+	"github.com/pingcap/tidb/util/codec"
```
Comment (3pointer) [inline pkg/restore/import.go]: CI failed, seems should use pd [codec](https://github.com/pingcap/br/pull/137/files#diff-eae46bd20abe35cad68c49bd179a3728R14)

Comment (3pointer) [reply pkg/restore/import.go]: but the codec in pd and tidb seems no different

Comment (overvenus) [reply pkg/restore/import.go]: Turns out I exchange `DecodeBytes` arguments by mistake.

**Pull request merged**